pkg/storage/snapshot/restore.go

```diff
@@ -111,27 +111,27 @@ func (ctrl *VMRestoreController) updateVMRestore(vmRestoreIn *snapshotv1.Virtual
 	logger := log.Log.Object(vmRestoreIn)
```

Comment:
Pretty complicated commit here, I think it would best to maybe break this down and only make changes that are absolutely necessary to address the underlying issue. A minimally viable fix. Any other refactorings/cleanup can be done later.

Also a description of why the tests are flaking would be helpful